packages/admin-ui/src/table/components/table-cell.tsx

```diff
@@ -29,12 +30,30 @@ export const TableCell = createComponent<'div', CellOptions>((props) => {
 
   const isLastFixedColumn = isFixed && lastFixedColumn?.id === column?.id
 
+  const isScrolled = () => {
+    if (!isLastFixedColumn) {
+      return false
+    }
+
+    const { hasHorizontalScroll } = useVerticalScroll()
+
+    return hasHorizontalScroll
+  }
+
   const ref = useRef<HTMLDivElement>(null)
 
   useEffect(() => {
-    if (ref.current && isFixed) {
-      ref.current.style.left = `${ref.current.offsetLeft}px`
+    if (!tableRef?.current || !ref?.current || !isFixed) {
+      return
     }
+
+    const isBodyCell = ref.current.getAttribute('role') === 'cell'
+
+    const table = tableRef.current
+
+    ref.current.style.left = isBodyCell
+      ? `${ref.current.offsetLeft - table.offsetLeft}px`
+      : `${ref.current.offsetLeft}px`
```

Comment:
I'm not sure if i get what this does exactly

Reply:
I will put a comment to explain it  ;)